requirements.txt

```diff
@@ -2,3 +2,4 @@ cosl
 jsonschema >=4.19,<4.20
 ops >= 2.6
 pydantic==2.9.2
+websockets < 14.0
```

Comment:
Isn't this only a requirement for the integration tests (python-libjuju)? If yes, it should better be placed in tox.ini